src/test/java/org/springframework/data/hazelcast/repository/query/QueryIT.java

```diff
@@ -831,4 +831,84 @@ public void findByLastnameListenableFuture()
         assertThat(result.get(), containsInAnyOrder(person1, person2));
     }
     
+    @Test
+    public void countDistinctPersonByFirstname() {
+    	//adding a duplicate Person with different id
+    	Person p = new Person();
+    	p.setId("1970");
+    	p.setFirstname("George");
+    	p.setLastname("Scott");
+    	this.personMap.put("1001", p);
+    	
+        final Long distinctPersons = this.personRepository.countDistinctLastnameByFirstname("George");
```

Comment:
What does `Lastname` in `countDistinctLastnameByFirstname` change here?

I believe `countDistinctLastnameByFirstname` and `countDistinctByFirstname` are effectively the same - worth double-checking and preparing a test case. I know it might sound counter-intuitive